acme_tiny.py

```diff
@@ -41,67 +46,57 @@ def _b64(b):
     thumbprint = _b64(hashlib.sha256(accountkey_json.encode('utf8')).digest())
 
     # helper function make signed requests
-    def _send_signed_request(url, payload):
+    def _send_signed_request(url, payload, return_codes, error_message):
         payload64 = _b64(json.dumps(payload).encode('utf8'))
         protected = copy.deepcopy(header)
         protected["nonce"] = urlopen(CA + "/directory").headers['Replay-Nonce']
         protected64 = _b64(json.dumps(protected).encode('utf8'))
-        proc = subprocess.Popen(["openssl", "dgst", "-sha256", "-sign", account_key],
-            stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        out, err = proc.communicate("{0}.{1}".format(protected64, payload64).encode('utf8'))
-        if proc.returncode != 0:
-            raise IOError("OpenSSL Error: {0}".format(err))
-        data = json.dumps({
-            "header": header, "protected": protected64,
-            "payload": payload64, "signature": _b64(out),
+        signed_request = json.dumps({
+            "header": header, "protected": protected64, "payload": payload64,
+            "signature": _b64(_openssl("dgst", ["-sha256", "-sign", account_key],
+                communicate="{0}.{1}".format(protected64, payload64).encode("utf8")))
         })
         try:
-            resp = urlopen(url, data.encode('utf8'))
-            return resp.getcode(), resp.read()
+            resp = urlopen(url, signed_request.encode("utf8"))
+            code, result = resp.getcode(), resp.read()
         except IOError as e:
```

Comment:
Ah yes, you are right, that's Python 3 urllib specific.

urllib Python2 documentation, explicitly [says that's an IOError is returned](https://docs.python.org/2/library/urllib.html#urllib.urlopen).

Although, Python 2 contains urllib2 and [there urllib2.HTTPError is defined](https://docs.python.org/2/library/urllib2.html#urllib2.HTTPError) as in Python 3, but not in the same module (urllib2 instead of urllib.error).

I've just seen in the header that acme-tiny uses already urrlib2 for Python2.
I'll try to see if we can make excepts on one of these two HTTPError.